### PR TITLE
fix: null notifications

### PR DESCRIPTION
**Pull request by rebelchris**
There's an [issue](https://github.com/dailydotdev/daily/issues/1488) where if you bookmark shared post without title it will show `null`.
This was actually a deeper lying issue in notification generation.

So decided to fallback in several places to sharedPost.title if it exists.

Screenshot 2024-10-11 at 14 42 33

### CODE REVIEW

**Comment by rebelchris** [comment]
Maybe we should run manual update query in DB to fix notification_v2.title
From:
Reading reminder! <b>null</b> 
To:
Reading reminder!
(just the quickest fix I guess)

Only 15 rows

File: src/notifications/generate.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export const notificationTitleMap: Record<
     `The collection "<b>${ctx.post.title}</b>" just got updated with new details`,
   dev_card_unlocked: () => 'DevCard unlocked!',
   post_bookmark_reminder: (ctx: NotificationPostContext) =>
-    `Reading reminder! <b>${ctx.post.title}</b>`,
+    `Reading reminder! <b>${ctx.post.title ? ctx.post.title : ctx.sharedPost?.title ? ctx.sharedPost?.title : undefined}</b>`,
```
Comment (capJavert) [inline src/notifications/generate.ts:99]: Can we do this with if-else, double ternary makes my head spin 😆

Comment (sshanzel) [reply src/notifications/generate.ts:99]: Yeah, I agree, maybe a generic function to fetch title that tries to fallback on possible ones.

File: src/notifications/generate.ts [hunk 2]
```diff
@@ -32,6 +32,20 @@ import { format } from 'date-fns';
 
 const systemTitle = () => undefined;
 
+const getPostOrSharedPostTitle = (
+  ctx: NotificationPostContext,
+): string | null | undefined => {
+  if (ctx.post.title && ctx.post.title?.length > 0) {
```
Comment (capJavert) [inline src/notifications/generate.ts:38]:
```suggestion
  if (ctx.post?.title?.length > 0) {
```

File: src/notifications/generate.ts [hunk 3]
```diff
@@ -32,6 +32,20 @@ import { format } from 'date-fns';
...
+    return ctx.post.title;
+  }
+
+  if (ctx.sharedPost?.title && ctx.sharedPost?.title?.length > 0) {
```
Comment (capJavert) [inline src/notifications/generate.ts:42]:
```suggestion
  if (ctx.sharedPost?.title?.length > 0) {
```

**Review by capJavert — APPROVED** [review]

File: src/notifications/generate.ts [hunk 4]
```diff
@@ -32,6 +32,20 @@ import { format } from 'date-fns';
 
 const systemTitle = () => undefined;
 
+const getPostOrSharedPostTitle = (
+  ctx: NotificationPostContext,
+): string | null | undefined => {
+  if (ctx.post?.title?.length > 0) {
```
Comment (rebelchris) [inline src/notifications/generate.ts:38]:
```suggestion
  if (ctx.post?.title?.length) {
```

File: src/notifications/generate.ts [hunk 5]
```diff
@@ -32,6 +32,20 @@ import { format } from 'date-fns';
...
+    return ctx.post.title;
+  }
+
+  if (ctx.sharedPost?.title?.length > 0) {
```
Comment (rebelchris) [inline src/notifications/generate.ts:42]:
```suggestion
  if (ctx.sharedPost?.title?.length) {
```

**Pull request merged**